_data-prepper/peer-forwarder.md

```diff
@@ -0,0 +1,158 @@
+* `drain_timeout`: A `Duration` representing the wait time for the peer forwarder to complete processing data before shutdown.
+
+# SSL configuration
+The SSL configuration for setting up trust manager for peer forwarding client to connect to other Data Prepper instances.
+
+## Optional SSL configuration
+
+* `ssl`: A `boolean` that enables TLS/SSL. Default value is `true`.
+* `ssl_certificate_file`: A `String` representing the SSL certificate chain file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Defaults to `config/default_certificate.pem` which is default certificate file. Read more about how the certificate file is generated [here](https://github.com/opensearch-project/data-prepper/tree/main/examples/certificates).
+* `ssl_key_file`: A `String` represents the SSL key file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Defaults to `config/default_private_key.pem` which is default private key file. Read more about how the private key file is generated [here](https://github.com/opensearch-project/data-prepper/tree/main/examples/certificates).
+* `ssl_insecure_disable_verification`: A `boolean` that disables the verification of server's TLS certificate chain. Default value is `false`.
+* `ssl_fingerprint_verification_only`: A `boolean` that disables the verification of server's TLS certificate chain and instead verifies only the certificate fingerprint. Default value is `false`.
+* `use_acm_certificate_for_ssl`: A `boolean` that enables TLS/SSL using certificate and private key from AWS Certificate Manager (ACM). Default is `false`.
+* `acm_certificate_arn`: A `String` represents the ACM certificate ARN. ACM certificate take preference over S3 or local file system certificate. Required if `use_acm_certificate_for_ssl` is set to `true`.
+* `acm_private_key_password`: A `String` that represents the ACM private key password which that will be used to decrypt the private key. If it's not provided, a random password will be generated.
+* `acm_certificate_timeout_millis`: An `int` representing the timeout in milliseconds for ACM to get certificates. Default value is `120000`.
+* `aws_region`: A `String` represents the AWS region to use `ACM`, `S3` or `AWS Cloud Map`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file` and `ssl_key_file` is `AWS S3` path or if `discovery_mode` is set to `aws_cloud_map`.
+
+```yaml
+peer_forwarder:
+  ssl: true
+  ssl_certificate_file: "<cert-file-path>"
+  ssl_key_file: "<private-key-file-path>"
+```
+
+# Authentication
+
+* `authentication`(Optional) : A `Map` that enables mTLS. It can either be `mutual_tls` or `unauthenticated`. Default value is `unauthenticated`.
+```yaml
+peer_forwarder:
+  authentication:
+    mutual_tls:
+```
+
+# Metrics
+
+Core Peer Forwarder introduces the following custom metrics and all the metrics are prefixed by `core.peerForwarder`.
```

Comment:
What are the custom metrics?

Reply:
@asifsmohammed Can we get some information on what the custom metrics are?

Reply:
The metrics documented below are the ones. They're categorized into Timer, Counter and Gauge metrics.

I think the use of word `custom` here is confusing. These are the only metrics provided by Peer Forwarder.